### DISCUSSION TITLE
Extracting Time Varying Contrail Properties as a Function of Time

### DISCUSSION THREAD

**Discussion by Calebsakhtar** [contrailcirrus/pycontrails#166] [Q&A] [answered]
Hi, my name is Caleb and I am a PhD student at the University of Cambridge working on contrail avoidance.

I am currently in need of extracting the time-varying properties of a specific contrail segment. I am looking to plot the contrail segment ice mass and ice number evolution with time. Is there any built-in way to do this in pycontrails?

I have been reading through the code and I suspect that a memory of the old contrail properties is not kept. If this is true, I would need to modify the code myself to store these values. Before undertaking this effort, I would like to know whether I am missing something. Further, if I implement a way to extract the evolution data, would this be a feature that you would consider including in pycontrails?

**Comment by Calebsakhtar** [comment]
@roger-teoh Sorry to bother you, just bumping this.

**Reply by roger-teoh** [reply, marked as answer]
Thank you for your question, @Calebsakhtar 

Yes, the CoCiP in pycontrails saves the contrail segment properties at time steps of your selected d_t_ (which has a default value of 30 minutes). You can change d_t_ to your preferences at pycontrails.models.cocip.cocip_params (dt_integration). This is the example block of codes that I used to extract the time-varying contrail properties at time steps of dt:

cocip = Cocip(met=met, rad=rad, params=params)
flight = cocip.eval(source=flight)   # One flight

if return_contrails:   # Returns the time-varying contrail properties
      return cocip.contrail
else:
      return flight

I hope this helps!

**Reply by Calebsakhtar** [reply]
Thanks so much! It’s greatly appreciated!
